### PR TITLE
Draft EIP: BLS12-381 Keystore

### PR DESCRIPTION
**Pull request by CarlBeek**
This is the proposed standard for BLS12-381 Keystores for use within Eth2 as well as by the larger blockchain industry. It has existed in its own repo for a while now, but this is a better place to enshrine it as a standard for now.

### CODE REVIEW

File: EIPS/eip-draft_bls_keystore.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+## A note on purpose
+
+This specification is designed not only to be an Ethereum 2.0 standard, but one that is adopted by the wider community who have adopted the BLS12-381 signature standard. It is therefore important also to consider the needs of the wider industry along with those specific to Ethereum. As a part of these considerations, it is the intention of the author that this standard eventually migrate to a more neutral repository in the future.
+
+## Motivation
+
+The secure storage and exchange of keys is a vital component of the user experience as people are expected to hold their own keys. It allows users to control access to individual keys and their use by applications.
+
+In Ethereum 1, [the Web2 Secret Storage Definition](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition) fulfills these requirements, however it is not perfectly suitable for these purposes moving forward. Specifically the problems with the existing standard are:
```
Comment (MrChico) [inline EIPS/eip-draft_bls_keystore.md]:
```suggestion
In Ethereum 1, [the Web3 Secret Storage Definition](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition) fulfills these requirements, however it is not perfectly suitable for these purposes moving forward. Specifically the problems with the existing standard are:
```

**Comment by mpetrunic** [comment]
Implementation of EIP in typescript: https://github.com/nodefactoryio/bls-keystore

**Comment by mcdee** [comment]
If implementations is a thing here is one in go: https://github.com/wealdtech/go-eth2-wallet-encryptor-keystorev4/

**Comment by axic** [comment]
@CarlBeek can you confirm you're happy with my tiny changes?

**Comment by CarlBeek** [comment]
>@CarlBeek can you confirm you're happy with my tiny changes?

@axic I am indeed, thank you!

Can I ask you to hold off just a little while longer on the merge though. I'm having an offline discussion which may have ramifications here.

**Comment by axic** [comment]
@CarlBeek did you progress with the discussion?

**Comment by CarlBeek** [comment]
@axic, they have concluded. Thank you for waiting. Please merge.

**Comment by axic** [comment]
Actually it seems the rendering of the tables is wrong: https://eips.ethereum.org/EIPS/eip-2335

**Comment by paulmillr** [comment]
Hey folks -- why isn't this using aes-gcm256?

**Comment by axic** [comment]
@paulmillr please leave this question on the discussion URL: https://github.com/ethereum/EIPs/issues/2339

**Comment by axic** [comment]
> Actually it seems the rendering of the tables is wrong: https://eips.ethereum.org/EIPS/eip-2335

@CarlBeek can you fix this?

**Comment by axic** [comment]
> Actually it seems the rendering of the tables is wrong: https://eips.ethereum.org/EIPS/eip-2335

@CarlBeek this still seems to be a problem.

**Pull request merged**